src/main/java/uk/nhs/hee/tis/trainee/sync/facade/PlacementEnricherFacade.java

```diff
@@ -417,4 +558,66 @@ private String getSiteName(Site site) {
   private String getSiteLocation(Site site) {
     return site.getData().get(SITE_LOCATION);
   }
+
+  /**
+   * Get the Placement Specialty ID from the placement.
+   *
+   * @param placement The placement to get the placmeent specialty id from.
+   * @return The placement specialty id.
+   *
+   *          Note: since only one primary specialty exists per placement,
+   *          we use placementId as placement specialty ID
+   */
+  private String getPlacementSpecialtyId(Placement placement) {
+    return placement.getTisId();
+  }
+
+  /**
+   * Get the Specialty ID from the placement specialty.
+   *
+   * @param placementSpecialty The placement specialty to get the specialty id from.
+   * @return The specialty id.
+   */
+  private String getSpecialtyId(PlacementSpecialty placementSpecialty) {
+    return placementSpecialty.getData().get(PLACEMENT_SPECIALTY_SPECIALITY_ID);
+  }
+
+  /**
+   * Get the Placement ID from the placement specialty.
+   *
+   * @param placementSpecialty The placement specialty to get the placement id from.
+   * @return The placement id.
+   */
+  private String getPlacementId(PlacementSpecialty placementSpecialty) {
+    return placementSpecialty.getData().get(PLACEMENT_SPECIALTY_PLACEMENT_ID);
+  }
+
+  /**
+   * Get the Placements for the specialty ID.
+   *
+   * @param id The specialty id to get the placements from.
+   * @return The placements.
+   */
+  private Set<Placement> getPlacementsBySpecialtyId(String id) {
+    Set<PlacementSpecialty> placementSpecialties = placementSpecialtyService
+        .findPlacementSpecialtiesBySpecialtyId(id);
+
+    return placementSpecialties.stream()
+        .map(Record::getData)
+        .map(data -> data.get(PLACEMENT_SPECIALTY_PLACEMENT_ID))
+        .map(placementService::findById)
+        .filter(Optional::isPresent)
+        .map(Optional::get)
```

Comment:
You could replace these with a `.flatMap(Optional.stream())` but not sure how performance compares.